verilog/analysis/checkers/port_name_suffix_rule_test.cc

```diff
@@ -0,0 +1,106 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verilog/analysis/checkers/port_name_suffix_rule.h"
+
+#include <initializer_list>
+
+#include "gtest/gtest.h"
+#include "common/analysis/linter_test_utils.h"
+#include "common/analysis/syntax_tree_linter_test_utils.h"
+#include "common/text/symbol.h"
+#include "verilog/CST/verilog_nonterminals.h"
+#include "verilog/analysis/verilog_analyzer.h"
+#include "verilog/parser/verilog_token_enum.h"
+
+namespace verilog {
+namespace analysis {
+namespace {
+
+using verible::LintTestCase;
+using verible::RunLintTestCases;
+
+// Tests that PortNameSuffixRule correctly accepts valid names.
+TEST(PortNameSuffixRuleTest, AcceptTests) {
+  const std::initializer_list<LintTestCase> kTestCases = {
+      {""},
+      {"module t (input logic name_i); endmodule;"},
+      {"module t (output logic abc_o); endmodule;"},
+      {"module t (inout logic xyz_io); endmodule;"},
+      {"module t (input logic [7:0] name_i); endmodule;"},
+      {"module t (output logic [2:0] abc_o); endmodule;"},
+      {"module t (inout logic [3:0] xyz_io); endmodule;"},
+      {"module t (input bit name_i); endmodule;"},
+      {"module t (output bit abc_o); endmodule;"},
+      {"module t (inout bit xyz_io); endmodule;"},
+      {"module t (input logic name_i,\n"
+       "output logic abc_o,\n"
+       "inout logic xyz_io,\n"
+       "input logic [7:0] namea_i,\n"
+       "output logic [2:0] abca_o,\n"
+       "inout logic [3:0] xyza_io,\n"
+       "input bit nameb_i,\n"
+       "output bit abcb_o,\n"
+       "inout bit xyzb_io);\n"
+       "endmodule;"},
+  };
```

Comment:
Suggest test cases:

* without direction (no `input`), in which case no suffix is checked.
* with `_ni`, `_no`, etc.
* a port declaration with multiple ids like `input logic x_in, y_in` (make sure the second one is checked as well)